packages/babel-preset-react-app/index.js

```diff
@@ -134,3 +136,29 @@ if (env === 'test') {
     // ]);
   }
 }
+
+module.exports = function(context, opts = {}) {
+  const moduleTypes = ['commonjs', 'amd', 'umd', 'systemjs'];
+  let modules = 'commonjs';
+
+  if (opts !== undefined) {
+    if (opts.modules !== undefined) modules = opts.modules;
+  }
+
+  if (modules === true) {
+    modules = 'commonjs';
+  }
+
+  if (modules !== false && moduleTypes.indexOf(modules) === -1) {
+    throw new Error(
+      "Preset es2015 'modules' option must be 'false' to indicate no modules\n" +
+        "or a module type which be be one of: 'commonjs' (default), 'amd', 'umd', 'systemjs'"
+    );
+  }
+
+  if (env !== 'test') {
+    presetConfig.presets[0][1].modules = modules;
```

Comment:
This is very fragile. Can we not rely on order here?

Reply:
I thought about it and came up with 2 solutions:

1. Create a helper function that calculates and returns the correct value for `modules` key.
2. **Wrap the preset code in a function and return it as `module.exports`**.

I went with the second approach. The `modules` option is calculated [at the top] fo later [used in the config]. As before, the config is stored in `presetConfig`.

[at the top]: https://github.com/zhirzh/create-react-app/blob/049e73ea077ad57ba508174c02848d0b9e03d3e9/packages/babel-preset-react-app/index.js#L12-L30

[used in the config]: https://github.com/zhirzh/create-react-app/blob/049e73ea077ad57ba508174c02848d0b9e03d3e9/packages/babel-preset-react-app/index.js#L131-L131